commands/chain.go

```diff
@@ -108,3 +112,54 @@ var chainLsCmd = &cmds.Command{
 		}),
 	},
 }
+
+var syncStatusCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Show status of chain sync operation.",
+	},
+	Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
+		syncStatus := GetPorcelainAPI(env).ChainSyncStatus()
+		if err := re.Emit(syncStatus); err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+var syncTipSetCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Sync and validate a chain.",
+	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("peerid", true, false, "Base58-encoded libp2p peer ID that a chain will be synced from"),
+		cmdkit.StringArg("cid", true, false, "CID of the TipSet to sync"),
+	},
+	Options: []cmdkit.Option{
+		cmdkit.BoolOption("trusted", "Trust the TipSet when syncing").WithDefault(true),
+		cmdkit.Uint64Option("height", "Height of the TipSet to sync").WithDefault(0),
```

Comment:
This is a bit weird. If we had a serious need for this functionality, I think we would want to avoid the user needing to specify the height, since we can fetch the tipset and inspect its height. The HandleNewTipSet method signature is the right shape for our internal use, but not for external consumption. I'd rather we do this properly, exposing an ergonomic endpoint, or not at all until we're prepared to do that work.

Reply:
A user only needs to specify a PeerID and CID, these other 2 are optional. But I hear ya, this is not really made for users. I personally would like to keep it as I'll probably use it occasionally, but if you have strong feelings about removing it I'll oblige.